google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/impl/PubSubApi.scala

```diff
@@ -140,7 +140,10 @@ private[pubsub] trait PubSubApi {
     for {
       request <- Marshal((HttpMethods.POST, uri, request)).to[HttpRequest]
       response <- doRequest(request, maybeAccessToken)
-      pullResponse <- Unmarshal(response).to[PullResponse]
+      pullResponse <- response.status match {
+        case StatusCodes.OK => Unmarshal(response).to[PullResponse]
+        case status => Future.failed(new RuntimeException(s"Unexpected publish response. Code: [${status}]"))
```

Comment:
Google PubSub sometimes returns quite nice explanations in body of response.
I was wondering about `Unmarshal.to[String]` for non OK responses and adding this information to exception message. WDYT?

Reply:
Yes, more error context is always a plus.